src/OpenCLScheduler.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+    This file is part of Leela Zero.
+    Copyright (C) 2018 Junhee Yoo
+
+    Leela Zero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Leela Zero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Leela Zero.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "GTP.h"
+#include "Random.h"
+#include "OpenCLScheduler.h"
+
+void OpenCLScheduler::initialize(const int channels) {
+    // multi-gpu?
+    if(!cfg_gpus.empty()) {
+        bool silent = false;
+        for(auto gpu : cfg_gpus) {
+            auto opencl = new OpenCL();
```

Comment:
Use make_unique<OpenCL>(); here, and move the init call up or use back() on the vector for it.